src/strat_engine/blockdev.rs

```diff
@@ -55,87 +55,109 @@ impl BlockDev {
                mda_size: Sectors,
                force: bool)
                -> EngineResult<BlockDev> {
-        let metadata = try!(fs::metadata(path));
+        Ok(try!(Self::batch_new(stratis_id, &[path], mda_size, force)).pop().unwrap())
+    }
 
-        if !metadata.file_type().is_block_device() {
-            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} is not a block device",
-                                                              path.display()))));
-        }
+    /// Initialize multiple blockdevs at once. This allows all of them
+    /// to be checked for usability before writing to any of them.
+    pub fn batch_new(stratis_id: &str,
```

Comment:
It would be helpful to add a header comment about the parameters or even rename ```stratis_id``` to ```pool_id``` since, AFAICT, that is what it is. Please do.